chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
+		if (this.subs.includes(user.userid)) this.subs.splice(this.subs.indexOf(user.userid), 1);
 	}
 
-	tallyVotes() {
-		let votes = this.parseVotes();
-		let max = 0;
-		let toKill = null;
-		for (let i in votes) {
-			if (votes[i].num > max) {
-				toKill = i;
-				max = votes[i].num;
-			} else if (votes[i].num === max) {
-				toKill = null;
-				max = votes[i].num;
-			}
-		}
-
-		if (toKill && toKill !== 'none') {
-			return this.players[toKill];
+	renamePlayer(user, oldUserid) {
+		if (user.userid === oldUserid) {
+			this.players[user.userid].name = user.name;
 		} else {
-			return false;
+			this.players[user.userid] = this.players[oldUserid];
+			this.players[user.userid].userid = user.userid;
+			this.players[user.userid].name = user.name;
+			delete this.players[oldUserid];
 		}
 	}
 
-	// Gamestate handling:
-
-	start() {
-		let rolesLeft = this.roles;
-
-		for (let i in this.players) {
-			let index = Math.floor(Math.random() * rolesLeft.length);
-			this.players[i].class = MafiaData.MafiaClasses[rolesLeft[index]];
-			rolesLeft.splice(index, 1);
-			if (!this.players[i].class.atStart) {
-				this.players[i].getRole();
-			}
-		}
-
-		this.room.send('|uhtmlchange|mafia' + this.room.gameNumber + 'pregame|<div class="infobox">The game has started!</div>');
-		this.allowRenames = false;
-		this.gamestate = 'initial';
-		this.progress();
+	removeBannedUser(user) {
+		// Player was banned, attempt to sub now
+		// If we can't sub now, make subbing them out the top priority
+		if (this.subs.length) return this.nextSub(user.userid);
+		this.reqestedSub.unshift(user.userid);
 	}
 
-	end(image, content) {
-		this.announcementWindow(image, content);
-
-		clearTimeout(this.timer);
-		this.timer = null;
-		this.room.game = null;
-		if (this.autoModchat && this.oldModchat && this.room.modchat === '+') {
-			this.room.modchat = this.oldModchat;
-		}
-		this.destroy();
+	forfeit(user) {
+		// Treat it as if the user was banned (force sub)
+		return this.removeBannedUser(user);
 	}
 
-	forceEnd() {
-		this.room.send("The game of mafia was forcibly ended.");
-
-		clearTimeout(this.timer);
-		this.timer = null;
-		this.room.game = null;
-		if (this.autoModchat && this.oldModchat && this.room.modchat === '+') {
-			this.room.modchat = this.oldModchat;
-		}
+	end(force) {
+		this.ended = true;
+		this.sendRoom(this.roomWindow(), {uhtml: true});
+		this.updatePlayers();
 		this.destroy();
+		return true;
 	}
+}
 
-	progress(early) {
-		if (!early) {
-			for (let i in this.players) {
-				if (!this.players[i].done) {
-					this.players[i].eliminate();
+exports.pages = {
+	mafia: function (query, user) {
+		if (!user.named) return Rooms.RETRY_AFTER_LOGIN;
+		if (!query.length) return `|deinit`;
+		let room = Rooms(query.shift());
+		if (!room || !room.users[user.userid] || !room.game || room.game.gameid !== 'mafia' || room.game.ended) return `|deinit`;
+		let isPlayer = user.userid in room.game.players, isHost = user.userid === room.game.hostid;
```

Comment:
``const``

Reply:
Also, PS convention is against declaring two variables on the same line like this.